content/js1/sprints/1/prep/index.md

```diff
@@ -8,4 +8,443 @@ backlog= 'Module-JS1'
+
+### Reusing instructions
+
+Let’s consider another scenario. Instead of adding or multiplying numbers, we’ll consider a number like `10.3`.  
+Given the number `10.3`, we could ask:
+
+> 🤔 "What is the nearest whole number to `10.3`?"
+
+The process of finding the nearest whole number to 10.3 is called **rounding**. So we could rephrase the previous question to be:
+
+> 🤔 "What does the number `10.3` **round** to?”
+
+Again we can use the Node REPL to round the number 10.3 to the nearest whole number. However, in this case, there is no arithmetical operator for rounding the number `10.3` in JavaScript.
+
+However, we will often want to carry out some sequence of operations **again and again**.
+For example, we will often want to round numbers again and again.
+
+So in this case we can use a **function**.
+
+{{<note title="Definition" type="info">}}
+
+A function is a reusable set of instructions.
+
+{{</note>}}
+
+`Math.round` is an example of a function. Recalling the definition, a function is a reusable set of instructions: `Math.round` contains instructions for rounding any number.
+
+Functions will often take **inputs** and then **apply their set of instructions to the inputs** to produce an **output**.
+
+So we can write `Math.round` in the Node REPL:
+
+![round-img](math.round-img.png)
+
+After we've hit enter to evaluate, we then get the following:
+
+![round-img](math.round-evaluated.png)
+
+So in this case, the REPL output `[Function: round]` is indicating that `Math.round` is a function.
+
+### 📲 Calling a function
+
+Once we have the name for a function, we need to get Node to read the function's instructions and execute them. Execution means the computer reads the instructions and carries them out. So we can write the following in the REPL:
+
+![round-img](math.round-number-10.3.png)
+
+Notice now we have `(` and `)` parentheses after the name of the function and a number inside the parentheses. The parentheses indicate that we are calling the function. The number inside the parentheses is the **input** we're passing to the function.
+
+{{<note title="Definition" type="info">}}
+
+**Calling a function** means telling the computer to read the function's instructions and carry out its instructions. When calling a function we can also pass inputs to the function.
+
+{{</note>}}
+
+`Math.round(10.3)` is a call expression: it can be thought of as saying:
+
+"apply the set of instructions for `Math.round` to the number `10.3`."
+
+If we type `Math.round(10.3)` then again we get the result 10.
+So we say that `Math.round(10.3)` outputs 10.
+
+{{<note title="Note" type="note">}}
+It's important to note that `Math.round` is **implementation opaque**.
+**Implementation opaque** means we can’t read the set of instructions for `Math.round`. This is because `Math.round` was authored by the developers that created NodeJS.
```

Comment:
This feels like we're emphasising something that we actually don't want trainees to know lots of depth - in other words, knowing what "implementation opaque" means is not relevant to getting a job.

I think the intention is to say "don't worry about the set of instructions for `Math.round`". But I think a lot of trainees would write down "implementation opaque" and try to memorise it as an important fact.

Could we just swap it out for something like:

```suggestion
We can't read the set of instructions for `Math.round` because it was written by the developers of NodeJS itself.
```

Reply:
Cool - I'm going to remove the idea from this point in the curriculum and refer to it in week 2.